apollo/resolvers/form/createForm.ts

```diff
@@ -0,0 +1,25 @@
+import { GQLContext } from '@/apollo/interfaces';
+import { Form } from '@/models/form';
+import { ApolloError } from 'apollo-server-micro';
+
+interface Args {
+  workspaceId: string;
+}
+
+export const createForm = async (
+  _: Record<string, never>,
+  { workspaceId }: Args,
+  { db, session }: GQLContext,
+) => {
+  const workspace = await db.workspaceRepo.get(workspaceId);
+
+  if (!workspace?.users.map(String).includes(session.user.id)) {
+    throw new ApolloError('Workspace not found', '404');
+  }
```

Comment:
I actually ended up going down a rabbit hole on this one, I created a custom directive that would always check to ensure they are a part of that workspace that you can add to any query or mutation you want, however, the reason I didn't like it is because it would have to make a call to get the workspace from the db, then make the call again once it got to the resolver since you can't pass the result from the directive to the resolver. Maybe we just make a util called `ensureUserWorkspace` that we just call

Reply:
I like the util idea